// ng2-table/components/table/paging.js
/// <reference path="../../tsd.d.ts" />
System.register(['angular2/angular2'], function(exports_1) {
    var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
        else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = (this && this.__metadata) || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var angular2_1;
    var Ng2TablePaging;
    return {
        setters:[
            function (angular2_1_1) {
                angular2_1 = angular2_1_1;
            }],
        execute: function() {
            Ng2TablePaging = (function () {
                function Ng2TablePaging() {
                    this.config = true;
                    this.tableChanged = new angular2_1.EventEmitter();
                }
                Ng2TablePaging.prototype.onChangePage = function (event) {
                    // Object.assign(this.config, event);
                    if (this.config) {
                        this.tableChanged.next({ paging: event });
                    }
                };
                Ng2TablePaging = __decorate([
                    angular2_1.Directive({
                        selector: '[ng2-table-paging]',
                        inputs: ['config: ng2TablePaging'],
                        outputs: ['tableChanged'],
                        host: {
                            '(pagechanged)': 'onChangePage($event)'
                        }
                    }), 
                    __metadata('design:paramtypes', [])
                ], Ng2TablePaging);
                return Ng2TablePaging;
            })();
            exports_1("Ng2TablePaging", Ng2TablePaging);
        }
    }
});
