// ng2-table/components/table/sorting.js
/// <reference path="../../tsd.d.ts" />
System.register(['angular2/angular2'], function(exports_1) {
    var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
        else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = (this && this.__metadata) || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var angular2_1;
    var Ng2ThSortable;
    return {
        setters:[
            function (angular2_1_1) {
                angular2_1 = angular2_1_1;
            }],
        execute: function() {
            Ng2ThSortable = (function () {
                function Ng2ThSortable() {
                    this.sortChanged = new angular2_1.EventEmitter();
                }
                Ng2ThSortable.prototype.onToggleSort = function (event) {
                    if (event) {
                        event.preventDefault();
                    }
                    if (this.config && this.column && this.column.sort !== false) {
                        switch (this.column.sort) {
                            case 'asc':
                                this.column.sort = 'desc';
                                break;
                            case 'desc':
                                this.column.sort = '';
                                break;
                            default:
                                this.column.sort = 'asc';
                                break;
                        }
                        this.sortChanged.next(this.column);
                    }
                };
                Ng2ThSortable = __decorate([
                    angular2_1.Directive({
                        selector: '[ng2-th-sortable]',
                        inputs: ['config: ng2ThSortable', 'column'],
                        outputs: ['sortChanged'],
                        host: {
                            '(click)': 'onToggleSort($event, $target)'
                        }
                    }), 
                    __metadata('design:paramtypes', [])
                ], Ng2ThSortable);
                return Ng2ThSortable;
            })();
            exports_1("Ng2ThSortable", Ng2ThSortable);
        }
    }
});
